tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1314,8 +1364,12 @@ tensorflow::Status ConvertConv2DHelper(
   VLOG(2) << "data_format: " << data_format;
   if (data_format == "NHWC") {
     // TODO(jie): transpose it back!
+    nvinfer1::ITensor* inp = output_tensor;
     output_tensor = ctx.TransposeTensor(output_tensor, {0, 2, 3, 1});
     TFTRT_RETURN_ERROR_IF_NULLPTR(output_tensor, node_def.name());
+    // Store input and output tensors so that quantization ranges from output of shuffle
+    // can be inferred backwards for input
+    ctx.quantization_infer_backwards_[output_tensor] = inp;
```

Comment:
I think this is equivalent to `ctx.CopyQuantizationRange(inp, output_tensor, false);`?

Also, why not do forward copy here? Actually in general how do we decide when to do forward copy and when to do backward? It seems we can always do both?

Reply:
Oh yes, it shouldve been CopyQuantizationRange.

Hmm I guess you can always do both. How about renaming CopyQuantizationRange function to something like `MarkQuantizationRangesAsInferrable(input_tensor, output_tensor)`
which adds two edges to the quantization_infer table:
input -> output
output -> input

These edges are then used to propagate ranges across tensors using the algorithm you described here https://github.com/tensorflow/tensorflow/pull/22788#discussion_r223259796

Reply:
That sounds good to me, thanks.